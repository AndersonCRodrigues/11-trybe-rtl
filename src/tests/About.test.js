import { screen } from '@testing-library/react';
import { About } from '../pages';
import renderWithRouter from '../renderWithRouter';

describe('<About />', () => {
  it('testa se a página contem informações sobre a pokédex', () => {
    renderWithRouter(<About />);

    const heading = screen.getByRole('heading', { name: 'About Pokédex', level: 2 });
    const paragrafo01 = screen.getByText(/This application simulates/i);
    const paragrafo02 = screen.getByText(/One can filter Pokémon/i);
    const image = screen.getByRole('img', { name: 'Pokédex' });
    const imageSrc = 'https://cdn2.bulbagarden.net/upload/thumb/8/86/Gen_I_Pok%C3%A9dex.png/800px-Gen_I_Pok%C3%A9dex.png';

    expect(heading).toBeInTheDocument();
    expect(paragrafo01).toBeInTheDocument();
    expect(paragrafo02).toBeInTheDocument();
    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute('src', imageSrc);
  });

  it('mesma snapshot', () => {
    const { container } = renderWithRouter(<About />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
