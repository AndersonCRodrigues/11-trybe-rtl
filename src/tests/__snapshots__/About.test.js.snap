// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> mesma snapshot 1`] = `
<section>
  <h2>
    About Pokédex
  </h2>
  <section>
    <p>
      This application simulates a Pokédex, a digital encyclopedia containing all Pokémon
    </p>
    <p>
      One can filter Pokémon by type, and see more details for each one of them
    </p>
    <img
      alt="Pokédex"
      class="pokedex-image"
      src="https://cdn2.bulbagarden.net/upload/thumb/8/86/Gen_I_Pok%C3%A9dex.png/800px-Gen_I_Pok%C3%A9dex.png"
    />
  </section>
</section>
`;
